static_src/components/page_header.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import style from 'cloudgov-style/css/cloudgov-style.css';
+
+import createStyler from '../util/create_styler';
+import AppStore from '../stores/app_store';
+import OrgStore from '../stores/org_store';
+import { pageKinds } from '../constants';
+import PageMetadataStore from '../stores/page_metadata_store';
+import SpaceStore from '../stores/space_store';
+
+
+// Prefer title set by props or PageMetadataStore, then fallback to guessing by
+// page kind
+function getTitle(props) {
+  const pageMetadata = PageMetadataStore.current;
+  let title = props.title || pageMetadata.title;
+  if (title) {
+    return title;
+  }
+
+  // TODO this seems strange that the page header needs to know about specific
```

Comment:
Using specialized page header components could work, but how would `main.js` specify the pageheader? `main.js` renders the container component, but the `PageHeader` is actually nested in the `MainContainer` > `Header` components. If I'm understanding this correctly, what would the logic in `MainContainer` look like to render this?

```javascript
// main.js
app() {
  ReactDOM.render(
    <MainContainer>
      <AppPageHeader />
      <AppContainer />
    </MainContainer>);
}
```

```javascript
// main_container.js
render() {
const pageHeader = findPageHeaderElement(this.props.children);
const content = withoutPageHeaderElement(this.props.children);
return (
  <Header>
    <SiteHeader />
    { pageHeader }
  </Header>
  <main>
    { content }
  </main>
);
}
```

What would `findPageHeaderElement` look like? Or is there a better way.